### ISSUE TITLE
Spotify Artist Notifications

### ISSUE THREAD

**Issue by pedrofracassi** [SwitchbladeBot/switchblade#1049]
Notify a channel when a new song/album by an artist becomes avaliable on Spotify

**Comment by metehus** [comment]
> becomes avaliable
wdym by that? when an album is released right?
how can we do that? requests every x times to check if theres a new album?

btw, we need to set up a notifications system, for things like that and make it super easy to make notifications messages and make configurations for channels & etc.

**Comment by pedrofracassi** [comment]
Yup! Paging @davipatury for the rescue

**Comment by pedrofracassi** [comment]
> wdym by that? when an album is released right?
Yeah, when something new gets added to the artist. Could be a single, an album, a podcast(?) or a video, cause I think spotify has those now

> how can we do that? requests every x times to check if theres a new album?
Yeah, I was thinking of something that would send requests every X seconds to check. We gotta check Spotify's ratelimit, tho. Maybe we could create some sort of microservice that uns separately from the bot and then sends the information over when new stuff is found?

**Comment by metehus** [comment]
> Maybe we could create some sort of microservice that runs separately

wanna docker it huh?

we could of course merge this, like, if two servers starts listening to the same artist, it runs together
we could do some extreme math, see the rate limit and balance the requests to fill it perfect in the time
